src/main/scala/com/nec/vectorengine/VeProcessTraits.scala

```diff
@@ -62,9 +62,18 @@ trait VeProcess {
 
   def stackAllocations: Map[Long, VeCallArgsStack]
 
+  def loadedLibraries: Map[String, LibraryReference]
+
   def allocate(size: Long): VeAllocation
 
-  def free(address: Long): Unit
+  /*
+    If `unsafe` is set to true, then the method will attempt to free the memory
+    location even if it's not recorded in the VeProcess' internal allocations
+    tracker.  This option is made available only for the purpose of releasing
+    memory allocated from the VE side inside function calls (i.e. not allocated
+    from the VH side through the `VeProcess` abstraction).
+  */
+  def free(address: Long, unsafe: Boolean = false): Unit
```

Comment:
The only way we can get into this situation is if we don't record that allocation when it is returned from the VE.  Or do we have any other situation where we have a really good reason to have that?

Reply:
I think this situation only occurs in `executeGrouping()`, where we have to manually delete memory that was not allocated by `VeProcess` itself, but by the code that `VeProcess` called, since currently, `VeProcess` only records allocations made through `veo_alloc_mem` calls.

Reply:
In pretty much every call to the VE the result is from a new allocation. This means, we want to register the results too. 

I don't think that is a blocker for this PR, but we should address that in a follow up.